File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& ht) {
        int maxm=0, lt=0, rt=ht.size()-1;
        while(lt<rt){
            maxm = max(maxm, (rt-lt)*min(ht[lt], ht[rt]));
            ht[lt]<ht[rt]?lt++:rt--;
        }
        return maxm;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int lt = 0, rt = height.length - 1, ans = 0;
        while (lt < rt) {
            if (height[lt] < height[rt])
                ans = Math.max(ans, (rt - lt) * height[lt++]);
            else
                ans = Math.max(ans, (rt - lt) * height[rt--]);
        }
        return ans;
    }
}